[ENH] Add variable roles to `ADMG` class

Ref #2278

HI, @ankurankan @Spinachboul 
I was writing test code for the `ADMG` class, and an error occurred during the commit process.

- error message
```sh
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

pgmpy/tests/test_base/test_ADMG.py:1:1: F401 'networkx as nx' imported but unused
pgmpy/tests/test_base/test_ADMG.py:338:9: F841 local variable 'separated' is assigned to but never used
```


The issue is that `nx` and `separated` are unused in the following code sections. How should I handle this?

Thanks.

- pgmpy/tests/test_base/test_ADMG.py
```py
import networkx as nx
```

```py
class TestADMGSeparation:
    """Test m-separation and m-connection."""

    ...

    def test_is_m_separated(self):
        """Test m-separation check."""
        # A and B should not be m-separated (they have bidirected edge)
        assert not self.admg.is_mseparated("A", "B")

        # Test with conditional set
        separated = self.admg.is_mseparated("A", "D", conditional_set={"C"})
        # This depends on the specific graph structure and d-separation rules

    ...

```

--- a/pgmpy/base/ADMG.py
+++ b/pgmpy/base/ADMG.py
@@ -103,7 +103,7 @@ class ADMG(_GraphRolesMixin, MultiDiGraph):
             latent = [latent] * len(nodes)
 
         for index in range(len(nodes)):
-            self.add_node(node=nodes[index], latents=latent[index])
+            self.add_node(node=nodes[index], latent=latent[index])
 
     def add_directed_edges(self, ebunch):
         """

--- a/pgmpy/tests/test_base/test_ADMG.py
+++ b/pgmpy/tests/test_base/test_ADMG.py
@@ -56,6 +56,33 @@ class TestADMGInitialization:
         assert admg.get_role_dict() == {"exposure": ["A", "B"], "outcome": ["C"]}
 
 
+class TestADMGNodeOperations:
+    """Test node addition and validation."""
+
+    def test_add_node(self):
+        """Test adding a single node."""
+        admg = ADMG()
+        admg.add_node("A")
+        admg.add_node("B")
+        admg.add_node("C", latent=True)
+        admg.add_node("D", latent=True)
+
+        assert set(admg.nodes()) == {"A", "B", "C", "D"}
+        assert set(admg.latents) == {"C", "D"}
+
+    def test_add_nodes_from(self):
+        """Test adding multiple nodes at once."""
+        admg = ADMG()
+        admg.add_nodes_from(["A", "B"])
+        admg.add_nodes_from(set(["C", "D"]))
+        admg.add_nodes_from(["E", "F"], latent=[False, True])
+        admg.add_nodes_from(["G", "H"], latent=True)
+        admg.add_nodes_from(set(["I", "J"]), latent=True)
+
+        assert set(admg.nodes()) == {"A", "B", "C", "D", "E", "F", "G", "H", "I", "J"}
+        assert set(admg.latents) == {"F", "G", "H", "I", "J"}
+
+
 class TestADMGEdgeOperations:
     """Test edge addition and validation."""
 

